models/standalone-price/src/transformers.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  Reference,
+  KeyReference,
+  TReference,
+  TReferenceGraphql,
+} from '@commercetools-test-data/commons';
+import { Transformer } from '@commercetools-test-data/core';
+import { type TCustomerGroup } from '@commercetools-test-data/customer-group';
+import type {
+  TStandalonePrice,
+  TStandalonePriceGraphql,
+  TStandalonePriceRest,
+} from './types';
+
+// Overloads
+function createCustomerGroupReference(
+  customerGroup: TCustomerGroup,
+  target: 'rest'
+): TReference<'customer-group'>;
+function createCustomerGroupReference(
+  customerGroup: TCustomerGroup,
+  target: 'graphql'
+): TReferenceGraphql<'customer-group'>;
+
+// Create a customer group reference depending on `id` or `key` field availability
+function createCustomerGroupReference(
```

Comment:
`key` is optional in `CustomerGroup` - I wanted consumers to get the type of reference they intended, so I opted to create a `KeyReference` if `key` was provided, otherwise create an `id` `Reference`